klickhouse/src/values/mod.rs

```diff
@@ -29,6 +30,93 @@ pub use ip::*;
 #[cfg(test)]
 mod tests;
 
+#[derive(Clone, Debug)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub enum MaybeString {
+    String(Arc<String>),
```

Comment:
Is it possible to use `Arc<str>` here? If we go via `Arc::from(std::str::from_utf8(x))`, it should avoid one extra allocation and indirection: arced `str` stores data directly in the arc.